GVFS/GVFS.Common/MountInternal.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GVFS.Common
+{
+    public class MountInternal
+    {
+        public string ServiceName { get; set; }
+        public bool StartedByService { get; set; }
+
+        public static MountInternal FromJson(string json)
+        {
+            return JsonConvert.DeserializeObject<MountInternal>(json);

```

Comment:
Can this throw if the json is not in the right format?

Reply:
@sanoursa yes it can throw.  Since it's internal I'm thinking I'd prefer the mount not to start if we get bad input.

Reply:
Yes, agreed that it should take down the process. Please just confirm that it is getting handled and logged as it tears down the process.